src/reflect/scala/reflect/internal/Symbols.scala

```diff
@@ -242,16 +243,15 @@ trait Symbols extends api.Symbols { self: SymbolTable =>
     // Rename as little as possible.  Enforce invariants on all renames.
     type TypeOfClonedSymbol >: Null <: Symbol { type NameType = Symbol.this.NameType }
 
-    // Abstract here so TypeSymbol and TermSymbol can have a private[this] field
-    // with the proper specific type.
```

Comment:
👍 for this change. I've tried variations of this myself before.

I think the justification in this removed comment was optimizing for the wrong thing.